src/NServiceBus.Core/Causation/MessageCausation.cs

```diff
@@ -9,7 +11,12 @@ public MessageCausation()
 
         protected internal override void Setup(FeatureConfigurationContext context)
         {
-            context.Pipeline.Register("AttachCausationHeaders", new AttachCausationHeadersBehavior(), "Adds related to and conversation id headers to outgoing messages");
+            if (!context.Settings.TryGet<Func<ConversationIdGeneratorContext, string>>(MessageCausationConfigurationExtensions.CustomConversationIdGeneratorKey, out var idGenerator))
+            {
+                idGenerator = _ => CombGuid.Generate().ToString();
+            }
+
+            context.Pipeline.Register("AttachCausationHeaders", new AttachCausationHeadersBehavior(idGenerator), "Adds related to and conversation id headers to outgoing messages");
```

Comment:
> Adds related to

Should we touch "related to" headers too?  Won't that break callbacks, etc? It's not needed for visualization in SI anyway, so I think this is a different case, and probably we don't want users to even think about touching it.

Reply:
This doesn't affect the related to header since its based on the message id